src/test/scala/firrtlTests/ConstantPropagationTests.scala

```diff
@@ -1531,22 +1531,30 @@ class ConstantPropagationIntegrationSpec extends LowTransformSpec {
     val input =
       s"""|circuit Foo:
           |  module Foo:
+          |    input in1: UInt<3>
           |    output out1: UInt<2>
           |    output out2: UInt<2>
           |    output out3: UInt<2>
+          |    output out4: UInt<4>
           |    out1 <= xor(SInt<2>(-1), SInt<2>(1))
           |    out2 <= or(SInt<2>(-1), SInt<2>(1))
           |    out3 <= and(SInt<2>(-1), SInt<2>(-2))
+          |    out4 <= asUInt(xor(asSInt(in1), SInt<4>(0)))
```

Comment:
This is a bit nitty, but this test can be further simplified (the `asUInt` isn't doing anything as `xor` returns a `UInt` type and the PR can just use an `SInt` directly instead of casting):
```suggestion
          |    input in1: SInt<3>
          |    output out1: UInt<2>
          |    output out2: UInt<2>
          |    output out3: UInt<2>
          |    output out4: UInt<4>
          |    out1 <= xor(SInt<2>(-1), SInt<2>(1))
          |    out2 <= or(SInt<2>(-1), SInt<2>(1))
          |    out3 <= and(SInt<2>(-1), SInt<2>(-2))
          |    out4 <= xor(in1, SInt<4>(0))
```